Content.Server/Physics/Controllers/ConveyorController.cs

```diff
@@ -131,11 +246,11 @@ public IEnumerable<(EntityUid, TransformComponent)> GetEntitiesToMove(ConveyorCo
 
                 // Yes there's still going to be the occasional rounding issue where it stops getting conveyed
                 // When you fix the corner issue that will fix this anyway.
-                var transform = Transform(entity);
-                var gridPos = gridMatrix.Transform(transform.WorldPosition);
+                var transform = xformQuery.GetComponent(entity);
+                var gridPos = gridMatrix.Transform(_transform.GetWorldPosition(transform, xformQuery));
                 var gridAABB = new Box2(gridPos - 0.1f, gridPos + 0.1f);
```

Comment:
When would a conveyor convey something not parented to the same grid?

If conveyors ahve to be anchored to work, I imagine you could just work in terms of local coordinates
which would get rid of the matrix transforms, world positions and container checks, both here and in ` Convey()`